app/mailers/application_mailer.rb

```diff
@@ -42,4 +42,10 @@ def group_mail(users, options)
   def set_locale_for(user)
     FastGettext.set_locale(user.locale.blank? ? "en" : user.locale)
   end
+
+  def set_url
+    unless (@url = URI.parse(Setting[:foreman_url])).present?
+      raise ":foreman_url is not set, please configure in the Foreman Web UI (More -> Settings -> General)"
```

Comment:
How does this occur?  foreman_url isn't a setting that accepts a blank value AFAIK.

(Administer, not More.)